lms/djangoapps/course_home_api/progress/v1/views.py

```diff
@@ -122,11 +127,26 @@ def get(self, request, *args, **kwargs):
 
         enrollment_mode, _ = CourseEnrollment.enrollment_mode_for_user(request.user, course_key)
 
-        course_grade = CourseGradeFactory().read(request.user, course)\
```

Comment:
Might look slightly more gross, but have you considered looking into `course_grade.course_data`? It looks like that might contain a lot of the data you want (has_scheduled_content for example).

Reply:
it looks like this wouldn't work because the get_course_blocks call within course grade doesn't include the include_has_scheduled_content=True parameter